src/core/storage/fileio/s3_api.cpp

```diff
@@ -111,20 +114,57 @@ bool bucket_name_valid(const std::string& bucket_name) {
 }
 
 
-std::string string_from_s3url(const s3url& parsed_url) {
-  std::string ret = "s3://" + parsed_url.access_key_id + ":"
-      + parsed_url.secret_key + ":";
+} // anonymous namespace
+
+S3Client init_aws_sdk_with_turi_env(s3url& parsed_url) {
+  // s3 client config
+  // DefaultCredentialProviderChain
+  Aws::Client::ClientConfiguration clientConfiguration;
+
+  // a little bit too long, anyway
+  clientConfiguration.requestTimeoutMs = 5 * 60000;
```

Comment:
What is the reasoning behind these values?  If there are situations where they could be adjusted, we should make them global variables (search for REGISTER_GLOBAL).

Reply:
Update:  Ah, I see, they are there before.  Ideally I think they should be global variables still.

Reply:
I thought about it. might be a good-first-issue for the new comer.